### ISSUE TITLE
Insert Random Data into Tables Using mysql_random_data_load

### ISSUE THREAD

**Issue by qwea456** [Percona-Lab/mysql_random_data_load#9]
what is error?

INFO[0000] cannot get table film struct: cannot read trigger: sql: Scan error on column index 5: unsupported Scan, storing driver.Value type into type *time.Time

**Comment by guriandoro** [comment]
Hi!

Can you send us the exact command used, table definition (SHOW CREATE TABLE) and mysql version you are trying it on? Also, trigger definitions, if any.


Thanks.

**Comment by qwea456** [comment]
Server version: 5.7.18-log MySQL Community Server (GPL)


[root@DB1 ~]# mysql_random_data_loader sakila film 10000 --host=127.0.0.1 --port=3306 --user=root --password=123
INFO[0000] cannot get table film struct: cannot read trigger: sql: Scan error on column index 5: unsupported Scan, storing driver.Value type into type *time.Time


mysql> use sakila
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

mysql> show create table film;
+-------
| film  | CREATE TABLE `film` (
  `film_id` smallint(5) unsigned NOT NULL AUTO_INCREMENT,
  `title` varchar(255) NOT NULL,
  `description` text,
  `release_year` year(4) DEFAULT NULL,
  `language_id` tinyint(3) unsigned NOT NULL,
  `original_language_id` tinyint(3) unsigned DEFAULT NULL,
  `rental_duration` tinyint(3) unsigned NOT NULL DEFAULT '3',
  `rental_rate` decimal(4,2) NOT NULL DEFAULT '4.99',
  `length` smallint(5) unsigned DEFAULT NULL,
  `replacement_cost` decimal(5,2) NOT NULL DEFAULT '19.99',
  `rating` enum('G','PG','PG-13','R','NC-17') DEFAULT 'G',
  `special_features` set('Trailers','Commentaries','Deleted Scenes','Behind the Scenes') DEFAULT NULL,
  `last_update` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`film_id`),
  KEY `idx_title` (`title`),
  KEY `idx_fk_language_id` (`language_id`),
  KEY `idx_fk_original_language_id` (`original_language_id`),
  CONSTRAINT `fk_film_language` FOREIGN KEY (`language_id`) REFERENCES `language` (`language_id`) ON UPDATE CASCADE,
  CONSTRAINT `fk_film_language_original` FOREIGN KEY (`original_language_id`) REFERENCES `language` (`language_id`) ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=1001 DEFAULT CHARSET=utf8 |
+-------
1 row in set (0.00 sec)




mysql> show triggers;
+----------------------+--------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+----------------+----------------------+----------------------+--------------------+
| Trigger              | Event  | Table    | Statement                                                                                                                                                                                                                                                                                                              | Timing | Created                | sql_mode                                                                                                                                             | Definer        | character_set_client | collation_connection | Database Collation |
+----------------------+--------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+----------------+----------------------+----------------------+--------------------+
| customer_create_date | INSERT | customer | SET NEW.create_date = NOW()                                                                                                                                                                                                                                                                                            | BEFORE | 2018-01-16 11:12:59.78 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
| ins_film             | INSERT | film     | BEGIN
    INSERT INTO film_text (film_id, title, description)
        VALUES (new.film_id, new.title, new.description);
  END                                                                                                                                                                                          | AFTER  | 2018-01-16 11:12:56.31 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
| upd_film             | UPDATE | film     | BEGIN
    IF (old.title != new.title) OR (old.description != new.description) OR (old.film_id != new.film_id)
    THEN
        UPDATE film_text
            SET title=new.title,
                description=new.description,
                film_id=new.film_id
        WHERE film_id=old.film_id;
    END IF;
  END | AFTER  | 2018-01-16 11:12:56.35 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
| del_film             | DELETE | film     | BEGIN
    DELETE FROM film_text WHERE film_id = old.film_id;
  END                                                                                                                                                                                                                                                     | AFTER  | 2018-01-16 11:12:56.35 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
| payment_date         | INSERT | payment  | SET NEW.payment_date = NOW()                                                                                                                                                                                                                                                                                           | BEFORE | 2018-01-16 11:13:00.38 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
| rental_date          | INSERT | rental   | SET NEW.rental_date = NOW()                                                                                                                                                                                                                                                                                            | BEFORE | 2018-01-16 11:13:00.75 | STRICT_TRANS_TABLES,STRICT_ALL_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,TRADITIONAL,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION | root@localhost | utf8                 | utf8_general_ci      | utf8_general_ci    |
+----------------------+--------+----------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+--------+------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------+----------------+----------------------+----------------------+--------------------+
6 rows in set (0.01 sec)

**Comment by percona-csalguero** [comment]
Cannot reproduce

```
mysql> select version();                                                                                                                                                                     
+-----------+                                                                                                                                                                                
| version() |                                                                                                                                                                                
+-----------+                                                                                                                                                                                
| 5.7.20    |
+-----------+
1 row in set (0.00 sec)
```

```
bin/mysql_random_data_loader --host=127.1 --port=3306 --user=root sakila film 100
INFO[2018-01-16T19:49:34-03:00] Starting
   0s [====================================================================] 100%
INFO[2018-01-16T19:49:34-03:00] 100 rows inserted
```
Original number of rows: 1000.
After running mysql_random_data_load:
```
mysql> select count(*) from film;
+----------+
| count(*) |
+----------+
|     1100 |
+----------+
1 row in set (0.00 sec)
```

**Comment by percona-csalguero** [comment]
Could you please try the latest version? (0.1.7)
https://github.com/Percona-Lab/mysql_random_data_load/releases/tag/0.1.7

**Comment by imp3142** [comment]
@percona-csalguero 
I was able to reproduce this issue:

```
$ cd ~/Work; ./mysql_random_data_loader tt Visitors 100 -P 5633 -umsandbox -pmsandbox 
INFO[0000] cannot get table Visitors struct: cannot read trigger: sql: Scan error on column index 5: unsupported Scan, storing driver.Value type <nil> into type *time.Time 
$ ./mysql_random_data_loader --version
Version   : 0.1.7
Commit    : 8ac3062019f7732e9525db0d870f3174d330ea78
Branch    : master
Build     : 2018-01-17T00:48:00-0300
Go version: go1.9.2

```
Created column in SHOW TRIGGERS statement can return NULL.

(emphasis mine)
> Created: The date and time when the trigger was created. This is a TIMESTAMP(2) value (with a fractional part in hundredths of seconds) for triggers created in MySQL 5.7.2 or later, **NULL for triggers created prior to 5.7.2.**

https://dev.mysql.com/doc/refman/5.7/en/show-triggers.html

**Comment by percona-csalguero** [comment]
Thanks for the error report and for debugging the error.
I've fixed this in v0.1.8.

Regards
